src/libraries/Common/src/Interop/SunOS/procfs/Interop.ProcFsStat.TryReadProcessStatusInfo.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class procfs
+    {
+        /// <summary>
+        /// Attempts to get status info for the specified process ID.
+        /// </summary>
+        /// <param name="pid">PID of the process to read status info for.</param>
+        /// <param name="processStatus">The pointer to processStatus instance.</param>
+        /// <returns>
+        /// true if the process status was read; otherwise, false.
+        /// </returns>
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_ReadProcessStatusInfo", SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern unsafe bool TryReadProcessStatusInfo(int pid, ProcessStatusInfo* processStatus);
+
+        internal struct ProcessStatusInfo
+        {
+            internal ulong ResidentSetSize;
+            // add more fields when needed.
+        }
+
+        internal static unsafe bool TryReadProcessStatusInfo(int pid, out ProcessStatusInfo statusInfo)
```

Comment:
The native implementation you added currently doesn't write to the status info on error, which means a caller could see statusInfo containing garbage. This function should respect the `out` and guarantee that statusInfo is initialized, either ensuring that the P/Invoke always does so or here default initializing it (`statusInfo = default;`) before calling into the native code.

Reply:
Added default initialization.